pkg/build/api/v1/types.go

```diff
@@ -52,8 +52,75 @@ type BuildSpec struct {
 	// scheduled in the system, that the build may be active on a node before the
 	// system actively tries to terminate the build; value must be positive integer
 	CompletionDeadlineSeconds *int64 `json:"completionDeadlineSeconds,omitempty" description:"optional duration in seconds the build may be active on a node before the system will actively try to mark it failed and kill associated containers; value must be a positive integer"`
+
+	// Info is a placeholder that can store multiple reasons why a build was
+	// triggered. It should cover for a future implementation of level triggered
+	// builds where you can multiple reasons why a build was triggered.
+	Info []*BuildTriggerInfo `json:"info,omitempty" description:"contains specific information about the triggers that fired a build"`
+}
+
+// BuildTriggerInfo contains information about why/who/what triggered a build.
+type BuildTriggerInfo struct {
+	// Details is a placeholder for  information about who triggered a build if
+	// it's either a user, a token or something else
+	Details BuildTriggerDetails `json:"details,omitempty" description:"generic information about a trigerred build and/or about the user that triggered it"`
+
+	// CIJobDetails is a generic placeholder that should store information about
+	// jobs from CI plugins that integrate with openshift and triggered a
+	// specific build
+	CIDetails CIJobDetails `json:"ciDetails,omitempty" description:"CI job information that triggered a build"`
+
+	// GitWebHookInfo holds information about a build that was triggered by a
+	// gitwebhook including its refs, branch, last commit id, etc. It also stores
+	// the type of plugin that triggered it, if it was a generic one, github,
+	// bitbucket, etc.
+	GitWebHook GitWebHookInfo `json:"gitWebHook,omitempty" description:"general information about a webhook that triggered this build"`
+
+	// ImageChangeBuildInfo - Stores info about the image that triggered a new
+	// build
+	ImageChange ImageChangeBuildInfo `json:"imageChange,omitempty" description:"information about an image change build trigger"`
+
+	// ConfigChangeBuildInfo - stores info about the last version of the config
+	// that triggered this build
+	ConfigChange ConfigChangeBuildInfo `json:"configChange,omitempty" description:"generic information about a configchange that triggered a build"`
+}
+
+// BuildTriggerDetails holds information about who/what triggered a build
+type BuildTriggerDetails string
+
+// CIJobDetails holds information about the CI job that triggered a
+// build
+type CIJobDetails struct {
+	// Type is type of the CI that started this. "jenkins", "travis", etc.
+	Type string `json:"type,omitempty" description:"type of CI that triggered the build"`
```

Comment:
it was just jenkins :) but then @mfojtik usgested that there might be more CIs in the future, so we i made a more general structure. 

however this also doesn't account for the fact that multiple CIs have different information. so that's also something that needs to be discussed.
@bparees what do you think?